### PR TITLE
Feature 26 customizable link blocks

### LINKED ISSUE #26: Customizable link blocks
As a user I'd like to have customizable link blocks on the front page

### PR DESCRIPTION
**Pull request by terolamminpaa**
resolves #26

### CODE REVIEW

File: src/components/pages/WelcomePage.tsx [hunk 1]
```diff
@@ -57,21 +61,24 @@ class WelcomePage extends React.Component<Props, State> {
 
     const api = ApiUtils.getApi();
 
-    const [posts, mainMenu, localeMenu] = await Promise.all(
+    const [posts, mainMenu, localeMenu, frontPageColumnPost] = await Promise.all(
```
Comment (jarinykanen) [inline src/components/pages/WelcomePage.tsx:64]:
```suggestion
    const [ posts, mainMenu, localeMenu, frontPageColumnPost ] = await Promise.all(
```

File: src/components/pages/WelcomePage.tsx [hunk 2]
```diff
@@ -90,16 +97,130 @@ class WelcomePage extends React.Component<Props, State> {
...
+  /**
+   * Returns columns
+   */
+  private columnsRendered = () => {
+    const { classes } = this.props;
+    const { columns } = this.state;
+    if (!columns) {
+      return;
+    }
+    return columns.map((column: any) => {
```
Comment (jarinykanen) [inline src/components/pages/WelcomePage.tsx:129]:
```suggestion
    const { columns } = this.state;
    
    if (!columns) {
      return;
    }
    
    return columns.map((column: any) => {
```

File: src/components/pages/WelcomePage.tsx [hunk 3]
```diff
@@ -90,16 +97,130 @@ class WelcomePage extends React.Component<Props, State> {
...
+
+    if (!frontPageColumnPost) {
+      return;
+    }
+
+    const renderedContent = frontPageColumnPost.content ? frontPageColumnPost.content.rendered : undefined;
+    if (!renderedContent) {
+      return;
+    }
+    ReactHtmlParser(renderedContent, { transform: this.transformContent });
```
Comment (jarinykanen) [inline src/components/pages/WelcomePage.tsx:156]:
```suggestion
    }
    
    ReactHtmlParser(renderedContent, { transform: this.transformContent });
```

File: src/components/pages/WelcomePage.tsx [hunk 4]
```diff
@@ -90,16 +97,130 @@ class WelcomePage extends React.Component<Props, State> {
...
+  }
+
+  /**
+   * transform html source content before it is rendered
```
Comment (jarinykanen) [inline src/components/pages/WelcomePage.tsx:160]:
```suggestion
   * Transform html source content before it is rendered
```

File: src/components/pages/WelcomePage.tsx [hunk 5]
```diff
@@ -90,16 +97,130 @@ class WelcomePage extends React.Component<Props, State> {
...
+
+    return convertNodeToElement(node, index, this.transformContent);
+  }
+
+  /**
+   * Parses columns from wp-columns
+   * 
+   * @param node DomElement
+   */
+  private wpColumnParsing = (node: DomElement, index: number) => {
```
Comment (jarinykanen) [inline src/components/pages/WelcomePage.tsx:184]: Add missing docs param

File: src/components/pages/WelcomePage.tsx [hunk 6]
```diff
@@ -90,16 +97,130 @@ class WelcomePage extends React.Component<Props, State> {
...
+    if (!node.children) {
+      return undefined;
+    }
+    const columnsDividedToTwo = node.children.filter(child => child.attribs && child.attribs.class === "wp-block-column");
```
Comment (jarinykanen) [inline src/components/pages/WelcomePage.tsx:188]:
```suggestion
    }
    
    const columnsDividedToTwo = node.children.filter(child => child.attribs && child.attribs.class === "wp-block-column");
```

File: src/components/pages/WelcomePage.tsx [hunk 7]
```diff
@@ -90,16 +97,130 @@ class WelcomePage extends React.Component<Props, State> {
...
+    columnsDividedTwiceToTwo.forEach(item => {
+      if (item.children) {
+        columnsDividedToFour = [ ...columnsDividedToFour, ...item.children.filter(child => child.attribs && child.attribs.class === "wp-block-column") ];
+      }
+    });
+    const nodes = columnsDividedToFour.map(item => {
+      const converted = convertNodeToElement(item, index, this.transformContent);
+      const children = converted.props.children;
+      return [...children];
+    });
```
Comment (jarinykanen) [inline src/components/pages/WelcomePage.tsx:205]:
```suggestion

    let columnsDividedToFour: DomElement[] = [];
    columnsDividedTwiceToTwo.forEach(item => {
      if (item.children) {
        columnsDividedToFour = [ ...columnsDividedToFour, ...item.children.filter(child => child.attribs && child.attribs.class === "wp-block-column") ];
      }
    });
    
    const nodes = columnsDividedToFour.map(item => {
      const converted = convertNodeToElement(item, index, this.transformContent);
      const children = converted.props.children;
      return [...children];
    });
```

File: src/components/pages/WelcomePage.tsx [hunk 8]
```diff
@@ -90,16 +97,130 @@ class WelcomePage extends React.Component<Props, State> {
...
+
+    return nodes;
+  }
+
+  /**
+   * get html element classes
+   *
+   * @param node DomElement
+   */
+  private getElementClasses = (node: DomElement): string[] => {
```
Comment (jarinykanen) [inline src/components/pages/WelcomePage.tsx:215]: Add return docs

File: src/components/pages/WelcomePage.tsx [hunk 9]
```diff
@@ -90,16 +97,133 @@ class WelcomePage extends React.Component<Props, State> {
...
+      }
+    }
+
+    return convertNodeToElement(node, index, this.transformContent);
+  }
+
+  /**
+   * Parses columns from wp-columns
+   * 
+   * @param node DomElement
```
Comment (terolamminpaa) [inline src/components/pages/WelcomePage.tsx:185]:
```suggestion
   * @param node DomElement
   * @param index current node index
```

File: src/components/pages/WelcomePage.tsx [hunk 10]
```diff
@@ -90,16 +97,137 @@ class WelcomePage extends React.Component<Props, State> {
...
+      return [...children];
+    });
+
+    return nodes;
+  }
+
+  /**
+   * get html element classes
+   *
+   * @param node DomElement
```
Comment (terolamminpaa) [inline src/components/pages/WelcomePage.tsx:220]:
```suggestion
   * @param node DomElement
   *
   * @returns string[]
```

File: src/components/pages/WelcomePage.tsx (hunk 5 above)
Comment (terolamminpaa) [reply src/components/pages/WelcomePage.tsx]: Added

File: src/components/pages/WelcomePage.tsx (hunk 8 above)
Comment (terolamminpaa) [reply src/components/pages/WelcomePage.tsx]: Added

**Pull request merged**